mentat/include_files.py

```diff
@@ -76,23 +76,21 @@ def abs_files_from_list(paths: list[Path], check_for_text: bool = True):
     return files_direct, files_from_dirs, invalid_paths
 
 
-def get_include_files(
-    paths: list[Path], exclude_paths: list[Path]
-) -> tuple[Dict[Path, CodeFile], list[str]]:
-    """Returns a complete list of text files in a given set of include/exclude Paths."""
+def get_exclude_files(exclude_paths: list[Path]) -> list[Path]:
+    """Get the set of excluded files from the given exclude paths."""
     git_root = GIT_ROOT.get()
     config = CONFIG_MANAGER.get()
 
-    paths = expand_paths(paths)
+    # Paths excluded by Session
     exclude_paths = expand_paths(exclude_paths)
-
     excluded_files_direct, excluded_files_from_dirs, _ = abs_files_from_list(
         exclude_paths, check_for_text=False
     )
     excluded_files = set(
         map(lambda f: f.path, excluded_files_direct | excluded_files_from_dirs)
     )
 
+    # Paths excluded by config
     glob_excluded_files = set(
```

Comment:
Right now, the config's glob_excluded_files only applied to files added by directory (so you can put in in *.json for example in your config to exclude all json files, but if you add a json file manually it will still work). In this PR you kind of merge both --exclude and the glob exclude so that they both only apply to directories, whereas before the --exclude argument would take precedence over manually included files; it might seem weird that we make the --exclude take preference over files you add individually one by one, but because zsh expands globs in the shell, if you make a glob include every file you're including is individual, so now there's no way to exclude files from an include glob (before, the --exclude argument was in place to do exactly this). I do think this is all overly convoluted, so I would be ok with this change (even though I think it's useful to be able to exclude parts of globs), but you need to change the help message for --exclude to make it clear that it only applies to manually added directories.

Reply:
Aha - that was actually a good explanation and I see what you mean. This might be a good excuse to just have a separate argument for auto-context, like 'ignore'.